csharp/ql/src/semmle/code/csharp/controlflow/internal/Splitting.qll

```diff
@@ -117,6 +131,7 @@ abstract class SplitKind extends TSplitKind {
    */
   int getListRank(ControlFlowElement cfe) {
     this.appliesTo(cfe) and
+    this.isEnabled(cfe.getEnclosingCallable()) and
```

Comment:
Why not `result < maxSplits()` here and get rid of `isEnabled` and `getRank`, or is the point that this would not be sufficient to prevent an exponential blow-up?

Reply:
That would work, but it would potentially be more restrictive than necessary. For example, if we have a method with two blocks, where 5 types of split apply to the first, and 5 others to the second (i.e., the splits from the first block do not overlap with those from the second), then my approach will allow for 5 splits in both blocks, whereas the simpler approach will only allow for a combined 5 splits across the two blocks. Does that make sense? :-)

Reply:
I vaguely understand!